### PR TITLE
refactor: ensure strict null check node,npm

### LINKED ISSUE #12869: Enforce strict null checks for versioning schemes
### Describe the proposed change(s).

- [x] `aws-machine-image`
- [x] `cargo`
- [x] `composer`
- [x] `docker`
- [x] `git`
- [x] `gradle` #13554
- [x] `hashicorp`
- [x] `helm`
- [x] `hex` #13394
- [x] `ivy`
- [x] `loose` #13454
- [x] `maven` #13332
- [x] `node` #13404
- [x] `npm` #13404
- [x] `nuget`
- [x] `pep440` #13409
- [x] `poetry`
- [x] `regex`
- [x] `rez` #13334
- [x] `ruby` #13278
- [x] `semver-coerced` #13161
- [x] `semver` #13161
- [x] `swift` #12850 
- [x] `ubuntu` #12849

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes:

- Fix strict null error.

## Context:

- Fixes #12869 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/versioning/node/schedule.ts [hunk 1]
```diff
@@ -11,5 +11,5 @@ interface NodeJsSchedule {
 export type NodeJsData = Record<string, NodeJsSchedule>;
 
 export const nodeSchedule: NodeJsData = JSON.parse(
-  dataFiles.get('data/node-js-schedule.json')
+  dataFiles.get('data/node-js-schedule.json') ?? ''
```
Comment (viceice) [inline lib/versioning/node/schedule.ts:14]:
```suggestion
  dataFiles.get('data/node-js-schedule.json')!
```
This should never undefined or null.

Not sure about `!`. Maybe needs a string cast.

Comment (RahulGautamSingh) [reply lib/versioning/node/schedule.ts:14]: Should I ignore it for the time-being?

Comment (RahulGautamSingh) [reply lib/versioning/node/schedule.ts:14]: non-null assertion is disabled in eslint config.

Comment (viceice) [reply lib/versioning/node/schedule.ts:14]: We can disable it here for now

Comment (RahulGautamSingh) [reply lib/versioning/node/schedule.ts]: Have added a different and returning an empty string.

`nodeSchedule` is used here, incase empty object is returned the function returns false, which seems the expected outcome incase we can't get version information.
https://github.com/renovatebot/renovate/blob/77708881845c9520e29e878db66830d88a5467c5/lib/versioning/node/index.ts#L32-L42

Comment (RahulGautamSingh) [reply lib/versioning/node/schedule.ts]: > `nodeSchedule` is used here, incase empty object is returned the function returns false, which seems the expected outcome incase we can't get version information.
> 
> https://github.com/renovatebot/renovate/blob/77708881845c9520e29e878db66830d88a5467c5/lib/versioning/node/index.ts#L32-L42
> 
> So implemented a solution that returns empty object incase the string imported from the file is null.

this causes coverage issues reverting to non-null assertion

Comment (RahulGautamSingh) [reply lib/versioning/node/schedule.ts]: above solution causes coverage issues which can only be solved by adding new test which uses a mock of `../../data-files.generated` . Reverting to using non-null assertion.

**Review by viceice — APPROVED** [review]

**Pull request merged**